### PR TITLE
Add a walking skeleton for the halight-based swaps

### PR DESCRIPTION
**Pull request by thomaseizinger**
This is just for people to follow along of what the mob is doing :)

Things to do before this is merged:

## Pull out

- [ ] Pull out tracing improvements of the Ethereum module  (#2379)
- [ ] Pull out extension of cnd's config file with lnd values (TS) (#2382)
- [ ] Pull out cnd's configuration handling (default lnd path) (#2378)
- [ ] Pull out changes to announce protocol (#2383)

## Before merge

- [x] Don't use `Actions` trait for han-halight but create one trait per action, i.e. `FundAction` with a fn `fund_action(&self) -> Option<Self::Output>` @thomaseizinger 
- [ ] Implement Swap Digest properly @D4nte 
- [x] Refactor `halight::create_watcher` to return a Stream instead (i.e. remove the dependency on `InvoiceStates` from this fn)

## Related issues

Resolves https://github.com/comit-network/comit-rs/issues/1954.
Resolves https://github.com/comit-network/comit-rs/issues/2297.

### CODE REVIEW

File: cnd/src/lnd.rs [hunk 1]
```diff
@@ -57,27 +57,68 @@ struct Payment {
...
+            Path(path) => {
+                let mut buf = Vec::new();
+                std::fs::File::open(path)?.read_to_end(&mut buf)?;
+                let certificate = reqwest::Certificate::from_pem(&buf)?;
+                Ok(LazyCertificate::Certificate(certificate))
+            }
+        }
+    }
+
+    pub fn certificate(&self) -> Result<&Certificate, Error> {
```
Comment (thomaseizinger) [inline cnd/src/lnd.rs]: @D4nte 

Instead of erroring here, you could just do the read logic then you don't have to care about the error case :)

Comment (D4nte) [reply cnd/src/lnd.rs]: Do you mean read the file in this function?

Yes it is what I have done initially however, I did not want to read the file "every time".
To not have to read the file everytime I have to have a `&mut self` and it just bubbled up too much.

Hence the current solution where I do not read the current file has often.

**Comment by thomaseizinger** [comment]
Replaced by https://github.com/comit-network/comit-rs/pull/2426.

**Comment by D4nte** [comment]
@thomaseizinger "Implement Swap Digest properly @D4nte" we have an issue tracking: #2385.

**Pull request closed without merging**